### PR TITLE
Profile methods

### PR DESCRIPTION
**Pull request by sandervanhooft**
Solves #332.

Adding:
- `$profile->enableMethod($methodId, $data) : Method`
- `$profile->disableMethod($methodId, $data) : null`

### CODE REVIEW

File: tests/Mollie/API/Endpoints/ProfileMethodEndpointTest.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            ),
+            new Response(
+                201,
+                [],
+                '{
+                    "resource": "method",
+                    "id": "bancontact",
+                    "description": "Bancontact",
+                    "image": {
+                        "size1x": "https://mollie.dev/external/icons/payment-methods/bancontact.png",
```
Comment (MollieRick) [inline tests/Mollie/API/Endpoints/ProfileMethodEndpointTest.php]: This is an error in the docs what we're fixing now. Please use www.mollie.com :-)

File: tests/Mollie/API/Endpoints/ProfileMethodEndpointTest.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                        "size2x": "https://mollie.dev/external/icons/payment-methods/bancontact%402x.png",
+                        "svg": "https://mollie.dev/external/icons/payment-methods/bancontact.svg"
+                    },
+                    "_links": {
```
Comment (MollieRick) [inline tests/Mollie/API/Endpoints/ProfileMethodEndpointTest.php]: We're adding `profile` to the link-set.

```
"profile": {
     "href": "https://api.mollie.com/v2/profiles/pfl_xxxx",
     "type": "application/hal+json"
}
```

**Comment by MollieRick** [comment]
@sandervanhooft Because this endpoint has a lot of impact on the merchant, we expect that developers also mention the prices when they use this endpoint. Can you also add the `?include=pricing` on the Methods API? :)

**Comment by sandervanhooft** [comment]
> @sandervanhooft Because this endpoint has a lot of impact on the merchant, we expect that developers also mention the prices when they use this endpoint. Can you also add the `?include=pricing` on the Methods API? :)

@MollieRick Noted! I'm working on the dev-friendly exception handling first.

File: tests/Mollie/API/Endpoints/ProfileMethodEndpointTest.php (hunk 1 above)
Comment (sandervanhooft) [reply tests/Mollie/API/Endpoints/ProfileMethodEndpointTest.php]: Done!

File: tests/Mollie/API/Endpoints/ProfileMethodEndpointTest.php (hunk 2 above)
Comment (sandervanhooft) [reply tests/Mollie/API/Endpoints/ProfileMethodEndpointTest.php]: Done, thanks!

**Comment by sandervanhooft** [comment]
> @sandervanhooft Because this endpoint has a lot of impact on the merchant, we expect that developers also mention the prices when they use this endpoint. Can you also add the `?include=pricing` on the Methods API? :)

@MollieRick Can you provide me with an example response?

**Comment by MollieRick** [comment]
> @MollieRick Can you provide me with an example response?

Sure. You'll find an example here in the [GET Methods doc](https://docs.mollie.com/reference/v2/methods-api/get-method).

A method can contain multiple pricing rates.

**Comment by sandervanhooft** [comment]
> > @MollieRick Can you provide me with an example response?
> 
> Sure. You'll find an example here in the [GET Methods doc](https://docs.mollie.com/reference/v2/methods-api/get-method).
> 
> A method can contain multiple pricing rates.

Ah ok, this is a different API, overlooked that. Will work on it in a separate PR/issue.

**Comment by sandervanhooft** [comment]
@MollieRick I think this PR is ready now. Will handle ApiException separately.

**Comment by Smitsel** [comment]
Good call to do the Exception handling in a different PR. It's better if @MollieRick takes a look at this one when he is back from vacation next week.

**Comment by MollieRick** [comment]
LGTM!

**Comment by sandervanhooft** [comment]
Thanks @Smitsel and @MollieRick ! 😄

**Pull request merged**